lib/query/index.js

```diff
@@ -0,0 +1,129 @@
+const { renderStatisticsResponse } = require('./render-statistics')
+const { renderPrecalulatedStatisticsResponse } = require('./render-precalculated-statistics')
+const { renderCountAndExtentResponse } = require('./render-count-and-exent')
+const { getGeometryTypeFromGeojson } = require('../helpers')
+
+function query (json, requestParams = {}) {
+  const {
+    features,
+    filtersApplied: {
+      all: skipFiltering
+    } = {}
+  } = json
+
+  const { f: requestedFormat } = requestParams
+
+  if (shouldRenderPrecalculatedData(json, requestParams)) {
+    return renderPrecalculatedData(json, requestParams)
+  }
+
+  const data = (skipFiltering || !features) ? json : filterAndTransform(json, requestParams)
+
+  if (shouldLogWarnings()) {
+    logWarnings(data, requestParams.f)
+  }
+
+  if (requestedFormat === 'geojson') {
+    return {
+      type: 'FeatureCollection',
+      features: data.features
+    }
+  }
+
+  return renderGeoservicesResponse(data, {
+    ...requestParams,
+    attributeSample: _.get(json, 'features[0].properties'),
+    geometryType: getGeometryTypeFromGeojson(json)
+  })
+}
+
+function shouldRenderPrecalculatedData ({ statistics, count, extent }, { returnCountOnly, returnExtentOnly }) {
+  if (statistics) {
+    return true
+  }
+
+  if (returnCountOnly === true && count !== undefined && returnExtentOnly === true && extent) {
+    return true
+  }
+
+  if (returnCountOnly === true && count !== undefined && !returnExtentOnly) {
+    return true
+  }
+
+  if (returnExtentOnly === true && extent && !returnCountOnly) {
+    return true
+  }
+
+  return false
+}
+
+function renderPrecalculatedData (data, { returnCountOnly, returnExtentOnly }) {
+  const { statistics, count, extent } = data
+
+  if (statistics) {
+    return renderPrecalulatedStatisticsResponse(data)
+  }
+
+  const retVal = {}
+
+  if (returnCountOnly) {
+    retVal.count = count
+  }
+
+  if (returnExtentOnly) {
+    retVal.extent = extent
+  }
+
+  return retVal
+}
+
+function shouldLogWarnings () {
+  return process.env.NODE_ENV !== 'production' && process.env.KOOP_WARNINGS !== 'suppress'
+}
+
+function renderGeoservicesResponse (data, params = {}) {
+  const {
+    outStatistics,
+    returnCountOnly,
+    returnExtentOnly,
+    returnIdsOnly,
+    outSR
+  } = params
+
+  if (returnCountOnly || returnExtentOnly) {
+    return renderCountAndExtentResponse(data, {
+      returnCountOnly,
+      returnExtentOnly,
+      outSR
+    })
+  }
+
+  if (returnIdsOnly) {
+    return renderIdsOnlyResponse(data)
+  }
+
+  if (outStatistics) {
+    return renderStatisticsResponse(data, params)
+  }
+
+  return renderFeaturesResponse(data, params)
+}
+
+function renderIdsOnlyResponse ({ features = [], metadata = {} }) {
+  const objectIdFieldName = metadata.idField || 'OBJECTID'
+
+  const objectIds = features.map(({ attributes }) => {
+    return attributes[objectIdFieldName]
+  })
+
+  return {
+    objectIdFieldName,
+    objectIds
+  }
+}
```

Comment:
Not necessary but seems like this is something you can move out into its own module similar to `renderStatisticsResponse` and others? Or maybe have all these "response" helpers exported from one file so there's only one import file?

Reply:
This one was fairly simply so it seemed like overkill to make it its own file.